CMakeLists.txt

```diff
@@ -2,6 +2,9 @@ cmake_minimum_required (VERSION 3.0)
 
 project(OpenCL VERSION 2.1)
 
+option (BUILD_SHARED_LIBS "Build shared libs" ON)
+option (TESTS "Enable building tests" ON)
```

Comment:
It's better to name it something like `OPENCL_BUILD_TESTS`, see:
* http://cgold.readthedocs.io/en/latest/tutorials/variables/cache.html#cache-name-recommendation